CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+  execute_process(COMMAND git checkout ${OSRM_VERSION} WORKING_DIRECTORY ${OSRM_DEPS_DIR})
+
+  # These will be set by the subproject
+  # LibOSRM_INCLUDE_DIRS
+  # LibOSRM_LIBRARIES
+  # LibOSRM_DEPENDENT_LIBRARIES
+  # LibOSRM_CXXFLAGS
+  set(ENABLE_MASON ON CACHE INTERNAL "Always use mason for building osrm-backend" FORCE)
+  add_subdirectory(${OSRM_DEPS_DIR})
+  include(${OSRM_DEPS_DIR}/third_party/mason/mason.cmake)
+
+  # FIXME we can remove this double versioning after the merge
+  set(MASON_TBB_VERSION "43_20150316")
+  mason_use(tbb VERSION ${MASON_TBB_VERSION})
+
+  # We need this because we depend on the shared libraries from TBB
+  foreach(libpath ${MASON_PACKAGE_tbb_LIBRARY_DIRS})
+    file(GLOB TBBGlob ${libpath}/*.*)
+    foreach(filepath ${TBBGlob})
+      get_filename_component(filename ${filepath} NAME)
+      set(targetfile "${CMAKE_SOURCE_DIR}/lib/binding/${filename}")
+      add_custom_command(OUTPUT ${targetfile}
+                         COMMAND ${CMAKE_COMMAND} -E copy ${filepath} ${CMAKE_SOURCE_DIR}/lib/binding/
+                         DEPENDS ${filepath})
+      list(APPEND OSRM_BINARIES "${targetfile}")
+    endforeach()
+  endforeach()
+
+  add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/lib/binding/osrm-extract
+                     COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:osrm-extract> ${CMAKE_SOURCE_DIR}/lib/binding/
+                     DEPENDS osrm-extract)
+  add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/lib/binding/osrm-contract
+                     COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:osrm-contract> ${CMAKE_SOURCE_DIR}/lib/binding/
+                     DEPENDS osrm-contract)
+  add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/lib/binding/osrm-routed
+                     COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:osrm-routed> ${CMAKE_SOURCE_DIR}/lib/binding/
+                     DEPENDS osrm-routed)
+  add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/lib/binding/osrm-datastore
+                     COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:osrm-datastore> ${CMAKE_SOURCE_DIR}/lib/binding/
+                     DEPENDS osrm-datastore)
+  list(APPEND OSRM_BINARIES "${CMAKE_SOURCE_DIR}/lib/binding/osrm-extract")
+  list(APPEND OSRM_BINARIES "${CMAKE_SOURCE_DIR}/lib/binding/osrm-contract")
+  list(APPEND OSRM_BINARIES "${CMAKE_SOURCE_DIR}/lib/binding/osrm-routed")
+  list(APPEND OSRM_BINARIES "${CMAKE_SOURCE_DIR}/lib/binding/osrm-datastore")
+else()
+  # These will be set by the find script
+  # LibOSRM_INCLUDE_DIRS
+  # LibOSRM_LIBRARIES
+  # LibOSRM_DEPENDENT_LIBRARIES
+  # LibOSRM_CXXFLAGS
+  find_package(LibOSRM REQUIRED)
+endif()
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/node_modules/node-cmake")
+
+set(NodeJS_CXX_STANDARD 14 CACHE INTERNAL "Use C++14" FORCE)
+set(NodeJS_DOWNLOAD ON CACHE INTERNAL "Download node.js sources" FORCE)
```

Comment:
This is what I stumbled upon a couple of month ago, too. Why do we have to make a HTTP request for C++ header when the NodeJs headers are available locally or - aren't they always available? For example node-cmake could not detect them installed via nvm at `~/.nvm/versions/node/vx.x.x/include/` but that looks like a node-cmake issue?